migrationTool/ams/AssetMigrator.cs

```diff
@@ -83,11 +86,22 @@ public override async Task MigrateAsync(CancellationToken cancellationToken)
         private async Task<AssetStats> MigrateAsync(MediaServicesAccountResource account, AsyncPageable<MediaAssetResource> assets, List<MediaAssetResource>? filteredList, ChannelWriter<double> writer, CancellationToken cancellationToken)
         {
             var stats = new AssetStats();
+
+            // Dictionary to restrict spawning of too many BlobServiceClients
+            var storageAccounts = new ConcurrentDictionary<string, BlobServiceClient>();
+
             await MigrateInParallel(assets, filteredList, async (asset, cancellationToken) =>
             {
-                var storage = await _resourceProvider.GetStorageAccountAsync(account, asset, cancellationToken);
+                var storageAccountName = asset.Data.StorageAccountName;
+                if (!storageAccounts.ContainsKey(storageAccountName))
+                {
+                    BlobServiceClient newBlobServiceClient = new BlobServiceClient(new Uri($"https://{storageAccountName}.blob.core.windows.net"), _credentials);
```

Comment:
This only works for Azure public cloud. Wont work for Azure sovereign clouds or China cloud. That is the reason to make an ARM API call to get the current endpoint.

Reply:
yes, that is a good catch.  

It is a good idea to create a ConcurrentDictionary to hold BlobServiceClient for each storage account, I wonder if it is worth to move this cache into AzureResourceProvider so that it can be applied to more scenarios, and no need to update code here inside MigrateAsync() method.

Reply:
I see, I am not familiar with sovereign/China clouds. I wonder if it might be worth pursuing with Azure support, asking them to relax the Azure storage account management API throttling limits temporarily?

Reply:
@srayan 

I wonder if this PR (https://github.com/Azure/azure-media-migration/pull/239) would be good to mitigate?  It caches BlobServiceClient.

Reply:
Sorry I had missed these messages. New PR looks good to me. I will go ahead and abandon this one